handlers/alarms-handler/src/alarmMappings.ts

```diff
@@ -107,33 +119,80 @@ const teamToAppMappings: Record<Team, string[]> = {
 	],
 };
 
-const buildAppToTeamMappings = (): Record<string, Team[]> => {
-	const mappings: Record<string, Team[]> = {};
-
-	for (const [team, apps] of Object.entries(teamToAppMappings)) {
-		for (const app of apps) {
-			const teams = mappings[app] ?? [];
-			teams.push(team as Team);
-
-			mappings[app] = teams;
-		}
+export class AlarmMappings {
+	constructor(mappings: Record<string, AppInfo[]> = teamToAppMappings) {
+		this.appToTeamMappings = this.buildAppToTeamMappings(mappings);
+		this.appToLogGroupOverrides = this.buildAppToLogGroupOverrides(mappings);
 	}
-	return mappings;
-};
-
-const appToTeamMappings: Record<string, Team[]> = buildAppToTeamMappings();
 
-export const getTeams = (appName?: string): Team[] => {
-	if (appName && appToTeamMappings[appName]) {
-		return appToTeamMappings[appName] as Team[];
-	}
+	private buildAppToTeamMappings = (
+		theMappings: Record<Team, AppInfo[]>,
+	): Record<string, Team[]> => {
+		const entries: Array<[Team, AppInfo[]]> = Object.entries(
+			theMappings,
+		) as Array<[Team, AppInfo[]]>; // `as` - hmm?
+
+		const teamToApp: Array<{ app: string; team: Team }> = entries.flatMap(
+			([team, appInfos]) =>
+				appInfos.map((appInfo) => {
+					const app = typeof appInfo === 'string' ? appInfo : appInfo.app;
+					return { team, app };
+				}),
+		);
+		const groups = groupBy(teamToApp, ({ app }) => app);
+
+		const mappings: Record<string, Team[]> = Object.fromEntries(
+			Object.entries(groups).map(([app, info]) => [
+				app,
+				info.map(({ team }) => team),
+			]),
+		);
```

Comment:
tempted to make an object mapValues function - if there isn't already such a thing?
```suggestion
		const mappings: Record<string, Team[]> = 
			mapValues(
			  groups,
			  (info) => info.map(({ team }) => team),
			)
```

Reply:
Same comment as above - I am not sure why this needs changing.

Reply:
this was because I changed it from a simple team->app mapping to a team->(app|{app, loglink[]}) mapping so I could extract the log links.
When I wrote the log link extractor I couldn't work out the equivalent to groupMap in typescript so I ended up with a mess!   I think the groupMap implementation is there in a later commit.  It made sense to update the existing mappings extractor to be consistent but now that it's not needed I will have a look to see if it's still adding value.